src/alvin0319/OffHand/OffHandInventory.php

```diff
@@ -27,75 +27,57 @@
 
 namespace alvin0319\OffHand;
 
-use BadMethodCallException;
-use pocketmine\entity\Entity;
 use pocketmine\inventory\BaseInventory;
-use pocketmine\item\Item;
 use pocketmine\network\mcpe\protocol\MobEquipmentPacket;
-use pocketmine\network\mcpe\protocol\InventoryContentPacket;
-use pocketmine\network\mcpe\protocol\types\inventory\ItemStackWrapper;
-use pocketmine\Player;
-
-class OffHandInventory extends BaseInventory{
-	public const SLOT_OFFHAND = 0;
-
-	/** @var Player */
+use pocketmine\network\mcpe\protocol\types\ContainerIds;
+use pocketmine\nbt\tag\ListTag;
+use pocketmine\item\Item;
+use pocketmine\entity\Human;
+use pocketmine\Server;
+use pocketmine\utils\MainLogger;
+class OffhandInventory extends BaseInventory{
+    /** @var Human */
 	protected $holder;
-
-	public function __construct(Player $holder){
-		parent::__construct([], 1);
+	public function __construct(Human $holder){
 		$this->holder = $holder;
+		parent::__construct();
 	}
-
-	public function getPlayer() : Player{
-		return $this->holder;
-	}
-
-	public function setSize(int $size) : void{
-		throw new BadMethodCallException("Cannot call setSize on OffHandInventory");
-	}
-
 	public function getName() : string{
-		return "OffHandInventory";
+		return "Offhand";
 	}
-
 	public function getDefaultSize() : int{
 		return 1;
 	}
-
-	public function setItemInOffHand(Item $item) : void{
-		$this->setItem(self::SLOT_OFFHAND, $item);
-		
-		$pk = new InventoryContentPacket();
-		$pk->windowId = $this->holder->getWindowId($this);
-		$pk->items = [ItemStackWrapper::legacy($item)];
-		$this->holder->sendDataPacket($pk);
-
-		$this->holder->getDataPropertyManager()->setByte(Entity::DATA_COLOR, Entity::DATA_TYPE_BYTE);
-
-		$this->sendMobEquipmentPacket($this->holder);
-
-		foreach($this->viewers as $player){
-			$this->sendMobEquipmentPacket($player);
-		}
+	public function setSize(int $size){
+		throw new \BadMethodCallException("Offhand can only carry one item at a time");
 	}
-
-	public function getItemInOffHand() : Item{
-		return $this->getItem(self::SLOT_OFFHAND);
+	public function setItem(int $index, Item $item, bool $send = true): bool{
+	    	parent::setItem($index, $item, $send);
+	    	$this->sendItem();
```

Comment:
This is unnecessary. because Inventory::onSlotChanged() called after Inventory->setItem()

Reply:
I added that because I wanted every player nearby was able to see the item change, so that not only the player that’s carrying the item